lib/plot_test.go

```diff
@@ -0,0 +1,68 @@
+package vegeta
+
+import (
+	"io/ioutil"
+	"testing"
+	"time"
+
+	lttb "github.com/dgryski/go-lttb"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestNewHTMLPlotSeries(t *testing.T) {
+	for _, tc := range []struct {
+		name string
+		in   Results
+		out  []*htmlPlotSeries
+	}{
+		{
+			name: "single attack",
+			in: Results{
+				{Attack: "foo", Timestamp: time.Unix(0, 0), Latency: time.Second},
+				{Attack: "foo", Timestamp: time.Unix(1, 0), Latency: time.Millisecond},
+			},
+			out: []*htmlPlotSeries{
+				{
+					Attack: "foo",
+					Began:  time.Unix(0, 0),
+					Ok: []lttb.Point{
+						{X: 0, Y: 0},
+						{X: 1, Y: 0},
+					},
+				},
+			},
+		},
+	} {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			diff := cmp.Diff(newHTMLPlotSeries(tc.in), tc.out)
+			if diff != "" {
+				t.Error("\n" + diff)
+			}
+		})
+	}
+}
+
+func BenchmarkHTMLPlot(b *testing.B) {
+	b.StopTimer()
+	// Build result set
+	rs := make(Results, 50000)
+	for began, i := time.Now(), 0; i < 50000; i++ {
+		rs[i] = Result{
+			Code:      uint16(i % 600),
+			Latency:   50 * time.Millisecond,
+			Timestamp: began.Add(time.Duration(i) * 50 * time.Millisecond),
+		}
+		if i%5 == 0 {
+			rs[i].Error = "Error"
+		}
+	}
+	// Start benchmark
+	b.ReportAllocs()
+	b.StartTimer()
+	for i := 0; i < b.N; i++ {
+		NewHTMLPlot("Vegeta Plot", 5000, rs).WriteTo(ioutil.Discard)
```

Comment:
Error return value of `(github.com/tsenart/vegeta/lib.HTMLPlot).WriteTo` is not checked